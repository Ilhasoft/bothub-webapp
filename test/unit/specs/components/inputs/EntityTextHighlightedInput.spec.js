import { shallowMount, createLocalVue } from '@vue/test-utils';
import EntityTextHighlightedInput from '@/components/inputs/EntityTextHighlightedInput';

const localVue = createLocalVue();

describe('EntityTextHighlightedInput.vue', () => {
  describe('simple', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallowMount(EntityTextHighlightedInput, {
        localVue,
        propsData: {},
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with valid entities', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallowMount(EntityTextHighlightedInput, {
        localVue,
        propsData: {
          value: 'hi',
          entities: [
            {
              start: 0,
              end: 2,
              entity: 'greet',
            },
          ],
        },
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('type in input', () => {
      let newValue;
      beforeEach(() => {
        const input = wrapper.find({ ref: 'input' });
        newValue = `${input.element.value} my name is douglas`;
        input.element.value = newValue;
        input.trigger('input');
      });

      test('emit input event', () => {
        expect(wrapper.emitted('input')).toBeDefined();
      });

      test('input value is same of component value', () => {
        expect(wrapper.emitted('input')[0][0]).toBe(newValue);
      });
    });

    describe('select text', () => {
      beforeEach(() => {
        const input = wrapper.find({ ref: 'input' });
        input.trigger('select');
      });

      test('emit selected', () => {
        expect(wrapper.emitted('selected')).toBeDefined();
      });
    });
  });
});
