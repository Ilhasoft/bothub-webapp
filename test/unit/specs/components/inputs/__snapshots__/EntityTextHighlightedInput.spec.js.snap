// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityTextHighlightedInput.vue simple renders correctly 1`] = `
<div class="control has-icons-right">
  <div class="highlighted">
    <div class="highlighted-item"><span class="highlighted-item-before"></span> <span class="highlighted-item-text entity-selected"></span></div>
    <input type="text" class="input">
  </div>
  <!---->
</div>
`;

exports[`EntityTextHighlightedInput.vue with valid entities renders correctly 1`] = `
<div class="control has-icons-right">
  <div class="highlighted">
    <div class="highlighted-item"><span class="highlighted-item-before"></span> <span class="highlighted-item-text entity-sunflower">hi</span></div>
    <div class="highlighted-item"><span class="highlighted-item-before"></span> <span class="highlighted-item-text entity-selected"></span></div>
    <input type="text" class="input">
  </div>
  <!---->
</div>
`;
